bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/ErrorCode.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command;
+
+import java.text.MessageFormat;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Error codes as specified in <a href="https://pjlink.jbmia.or.jp/english/data_cl2/PJLink_5-1.pdf">[PJLinkSpec]</a> chapters
+ *  2.3. Set commands
+ *  2.4. Get commands
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public enum ErrorCode {
+    UNDEFINED_COMMAND("Undefined command", "ERR1"),
+    OUT_OF_PARAMETER("Out of parameter", "ERR2"),
+    UNAVAILABLE_TIME("Unavailable time", "ERR3"),
+    DEVICE_FAILURE("Projector/Display failure", "ERR4");
+
+    private String text;
+    private String code;
+
+    private ErrorCode(String text, String code) {
+        this.text = text;
+        this.code = code;
+    }
+
+    public static @Nullable ErrorCode getValueForCode(String code) {
+        for (ErrorCode result : ErrorCode.values()) {
+            if (result.code.equals(code.toUpperCase())) {
+                return result;
+            }
+        }
+
+        return null;
+    }
+
+    public String getText() {
+        return this.text;
+    }
+
+    /**
+     * Checks if a the given code is an error code. Optionally, restrictCodesTo can restrict the allowed error codes (based on PJLink specification).
+     *
+     * @param code string to be checked for error codes
+     * @param restrictCodesTo list of expected error codes according to PJLink specification, can be null if all error codes (ERR1-ERR4) can be expected
+     * @throws ResponseException
+     */
+    public static void checkForErrorStatus(String code, @Nullable Set<ErrorCode> restrictCodesTo) throws ResponseException {
+        ErrorCode parsed = getValueForCode(code);
+        if (parsed != null && (restrictCodesTo == null || restrictCodesTo.contains(parsed))) {
```

Comment:
That's exactly what I expect. There are error codes ERR1-ERR4. For some commands, e.g. only ERR1-ERR3 are expected error codes, ERR4 would be a valid value (imagine a projector with name "ERR4", this would not be an error but the actual name.

Reply:
Null means that all error codes are to be expected, so it throws "more" exceptions then if I reduce the potential error codes to be expected.

Reply:
If you are sure, ok, my feeling was then wrong ;)